code/part-two/client/source/App.jsx

```diff
@@ -1,22 +1,86 @@
 import React from 'react';
 import { Link, Route, Switch } from 'react-router-dom';
 
+import { SignupLogin } from './SignupLogin';
+
 import { Collection } from './Collection';
 import { CollectionList } from './CollectionList';
 import { Moji } from './Moji';
 import { Offer } from './Offer';
 import { OfferList } from './OfferList';
 
 export class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      privateKey: this.props.privateKey,
+      publicKey: this.props.publicKey
+    }
+  }
+
+  set privateKey(key) {
+    this.setState({ privateKey: key });
+    localStorage.setItem('privateKey', key);
+  }
+
+  get privateKey() {
+    return this.state.privateKey 
+      ? this.state.privateKey
+      : localStorage.getItem('privateKey');
```

Comment:
can this just be `return this.state.privateKey || localStorage.getItem('privateKey')` to avoid a ternary?

(same for `get publicKey()` below?)